src/modules/function/declaration_utils.rs

```diff
@@ -6,23 +6,32 @@ use crate::utils::cc_flags::{CCFlags, get_ccflag_name};
 use crate::utils::context::Context;
 use crate::utils::function_interface::FunctionInterface;
 
-pub fn skip_function_body(meta: &mut ParserMetadata) -> (usize, usize, bool) {
+pub fn skip_function_body(
+    meta: &mut ParserMetadata,
+    declared_failable: bool,
+    returns_tok: &Option<Token>
+) -> Result<(usize, usize, bool), Failure> {
     let index_begin = meta.get_index();
     let mut is_failable = false;
     let mut scope = 1;
     while let Some(tok) = meta.get_current_token() {
         match tok.word.as_str() {
             "{" => scope += 1,
             "}" => scope -= 1,
-            "fail" => is_failable = true,
+            "fail" => {
+                if !declared_failable && returns_tok.is_some() {
+                    return error!(meta, Some(tok), "Functions that can fail must have a '?' after the type name");
```

Comment:
Let's change this line to:
```suggestion
                    return error!(meta, Some(tok), "Failable functions must have a '?' after the type name");
```
for consistency